src/rngs/adapter/reseeding.rs

```diff
@@ -220,6 +237,43 @@ impl<R, Rsdr> CryptoRng for ReseedingCore<R, Rsdr>
 where R: BlockRngCore + SeedableRng + CryptoRng,
       Rsdr: RngCore + CryptoRng {}
 
+
+// Fork protection
+//
+// We implement fork protection on Unix using `pthread_atfork`.
+// When the process is forked, we increment `RESEEDING_RNG_FORK_COUNTER`.
+// Every `ReseedingRng` stores the last known value of the static in
+// `fork_counter`. If the cached `fork_counter` is less than
+// `RESEEDING_RNG_FORK_COUNTER`, it is time to reseed this RNG.
+//
+// If reseeding fails, we don't deal with this by setting a delay, but just
+// don't update `fork_counter`, so a reseed is attempted a soon as possible.
+#[cfg(all(feature="std", unix, not(target_os="emscripten")))]
+static mut RESEEDING_RNG_FORK_COUNTER: u64 = ::core::u64::MAX;
+
+#[cfg(all(feature="std", unix, not(target_os="emscripten")))]
+extern fn forkhandler() {
+    unsafe { RESEEDING_RNG_FORK_COUNTER += 1; }
+}
+
+#[cfg(all(feature="std", unix, not(target_os="emscripten")))]
+fn register_fork_handler() {
+    if get_fork_counter() == ::core::u64::MAX {
+        unsafe {
+            RESEEDING_RNG_FORK_COUNTER = 0;
+            libc::pthread_atfork(None, None, Some(forkhandler));
+        }
+    }
+}
+#[cfg(not(all(feature="std", unix, not(target_os="emscripten"))))]
+fn register_fork_handler() {}
+
+#[cfg(all(feature="std", unix, not(target_os="emscripten")))]
+fn get_fork_counter() -> u64 { unsafe { RESEEDING_RNG_FORK_COUNTER } }
```

Comment:
Do you know why reading mutable static data is unsafe? Because it's not even guaranteed that the value read will equal the old or new value if concurrent with a write.

So can `get_fork_counter()` and `forkhandler()` be called concurrently? If so we should instead use `std::sync::atomic::AtomicUsize` (with relaxed ordering).